plugins/package-managers/python/src/main/kotlin/Poetry.kt

```diff
@@ -88,8 +88,6 @@ class Poetry(
         val packages = resultsForScopeName
             .flatMap { (_, results) -> results.packages }
             .toOrtPackages()
-            .distinctBy { it.id }
-            .toSet()
```

Comment:
commit-msg: To me "And if it really
occurs, it should be handled differently than omitting an arbitrary
package." seems mis-leading, because the following code in `toOrtPackages()` does a de-duplication per 
id aka. `"${it.name}:${it.version}"`:

```
groupBy { "${it.name}:${it.version}" }.mapTo(mutableSetOf()) {
...
val pkg = packages.first()
``` 

For clarity, it would be nice to change that code to: `disctinctBy { "${it.name}:${it.version}" } ` imo.

Reply:
I've rephrased the commit message to better explain why I still believe this code can safely be dropped in whole.